src/mom5/ocean_param/lateral/ocean_submesoscale.F90

```diff
@@ -1397,23 +1397,27 @@ subroutine compute_bldepth(Time, Thickness, Dens, T_prog, surf_blthick)
 
   ! inverse front length = f/(<N> H), with H=hblt and <N> ave buoyancy freq over hblt
   if(front_length_deform_radius) then 
-      do j=jsd,jed
-         do i=isd,ied
-            buoy_freq_ave(i,j)    = 0.0
-            mld_thickness         = epsln
-            front_length_inv(i,j) = front_length_const_inv 
-            if(kblt(i,j) >= min_kblt) then  
-                buoy_freq_ave(i,j)= epsln
-                mld_thickness     = epsln
-                do k=1,kblt(i,j)
-                   mld_thickness      = mld_thickness      + Thickness%dzt(i,j,k)  
-                   buoy_freq_ave(i,j) = buoy_freq_ave(i,j) - grav_rho0r*Thickness%dzt(i,j,k)*Dens%drhodz_zt(i,j,k)  
-                enddo
-                buoy_freq_ave(i,j)    = sqrt(abs(buoy_freq_ave(i,j))/mld_thickness) 
-                front_length_inv(i,j) = min(front_length_const_inv, coriolis_param(i,j)/(epsln+mld_thickness*buoy_freq_ave(i,j))) 
-            endif 
-         enddo
-      enddo
+     front_length_inv = front_length_const_inv 
+     wrk1_2d(:,:)         = epsln
+     where(kblt>=min_kblt)
```

Comment:
I love a good where more than most, but isn't it mostly redundant with the test at 1410, and the where at 1417?

Reply:
It's needed for reproducibility with older results and the diagnostic. You need to initialise `buoy_freq_ave` to `epsln` where you are going to do the averaging and zero elsewhere. As you point out the division by zero issue is already avoided.